rules_128tech/pylint_plugins/mock_patch_autospec_test.py

```diff
@@ -0,0 +1,281 @@
+"""Unit tests for the mock_patch_autospec plugin"""
+
+import contextlib
+from unittest import mock
+
+import astroid
+import astroid.node_classes
+import pytest
+from pylint import testutils
+
+from rules_128tech.pylint_plugins import mock_patch_autospec
+
+
+_MOCK_PATCH_TEST_CASES = [
+    pytest.param('("foo")', True, id="no_autospec",),
+    pytest.param('("foo", FakeFoo())', False, id="new_arg_specified",),
+    pytest.param('("foo", new=FakeFoo())', False, id="new_kwarg_specified",),
+    pytest.param('("foo", spec=FakeFoo)', False, id="spec_kwarg_specified",),
+    pytest.param('("foo", spec_set=FakeFoo)', False, id="spec_set_specified",),
+    pytest.param('("foo", autospec=True)', False, id="autospec_true",),
+    pytest.param('("foo", autospec=False)', False, id="autospec_false",),
+]
+
+
+_MOCK_PATCH_OBJECT_TEST_CASES = [
+    pytest.param('(time, "time")', True, id="no_autospec",),
+    pytest.param('(time, "time", FakeFoo())', False, id="new_arg_specified",),
+    pytest.param('(time, "time", new=FakeFoo())', False, id="new_kwarg_specified",),
+    pytest.param('(time, "time", spec=FakeFoo)', False, id="spec_kwarg_specified",),
+    pytest.param('(time, "time", spec_set=FakeFoo)', False, id="spec_set_specified",),
+    pytest.param('(time, "time", autospec=True)', False, id="autospec_true",),
+    pytest.param('(time, "time", autospec=False)', False, id="autospec_false",),
+]
+
+_MOCK_PATCH_DICT_TEST_CASES = [
+    pytest.param('("os.environ", FOO="bar")', id="kwargs",),
+    pytest.param('("os.environ", {"FOO": "bar"})', id="dict",),
+]
+
+
+class TestUnittestMockPatch(testutils.CheckerTestCase):
+
+    CHECKER_CLASS = mock_patch_autospec.MockPatchAutospecChecker
+
+    @pytest.fixture(
+        params=[
+            pytest.param(
+                """
+                {imports}
+
+                class FakeFoo:
+                    pass
+
+                @{func_name}{call_args}  #@
+                def test():
+                    pass
+                """,
+                id="decorator",
+            ),
+            pytest.param(
+                """
+                {imports}
+
+                class FakeFoo:
+                    pass
+
+                def test():
+                    with {func_name}{call_args}:  #@
+                        pass
+                """,
+                id="context_manager",
+            ),
+        ]
+    )
+    def source_template(self, request):
+        return request.param
+
+    @pytest.mark.parametrize(
+        "imports, func_name",
+        [
+            pytest.param(
+                "import unittest.mock", "unittest.mock.patch", id="fully_qualified"
+            ),
+            pytest.param(
```

Comment:
does `from unittest import mock as smth` work?

Reply:
Just checked and yes, since the lookup is checking the original name instead of the possible alias. I'll add a test case for this to be certain though